engine/runtime-instrument-runtime-server/src/main/java/org/enso/interpreter/instrument/runtime/server/RuntimeServerInstrument.java

```diff
@@ -120,12 +120,13 @@ protected void onCreate(Env env) {
       MessageEndpoint client =
           env.startServer(URI.create(RuntimeServerInfo.URI), this.handler.endpoint());
       if (client != null) {
+        env.registerService(this);
```

Comment:
Was there a race condition? Or is this a change in the logic where we only register this service when the `MessageEndpoint` is created?

Will not the infrastructure throw error when a service is defined by `@InstrumentRegistraiton` but not registered via `registerService` then?